### ISSUE TITLE
Sign plugins?

### ISSUE THREAD

**Issue by Bananeweizen** [checkstyle/eclipse-cs#489]
https://github.com/checkstyle/eclipse-cs/pull/478#pullrequestreview-1285887724 made me think of whether we could sign the Eclipse plugins. I've never done that with other open source plugins simply because of missing infrastructure and certificates. But since the checkstyle project has an HTTPS enabled website, I guess all the necessary certificates are generally available somewhere.

Signing eclipse plugins is done via the maven-jarsigner-plugin, so there is nothing eclipse special, AFAIK. Does someone from the core team know how to get something signed with the checkstyle certificates without me having them and without compromising those certificates while testing this? E.g. having some private github action or an own maven-plugin with hardcoded signing information etc.

Does the main checkstyle project do any signing of generated artifacts?

**Comment by romani** [comment]
Yes, it is requirements of upload to maven central. It is not complicated to create own key to sign.
@rnveach , can you share link to doc how you did this recently?

**Comment by romani** [comment]
Some doc that we created for signing:
https://github.com/sevntu-checkstyle/dsm-maven-plugin/wiki/How-to-config-GPG-and-sign-artifact-with-it

**Comment by Bananeweizen** [comment]
After reading that one and several other pages, I'm quite certain those are 2 different mechanisms. The signing for Maven central is based on GPG and independent of Java, while the Eclipse signing relies on the Java keystore and sign tool.
I also found that a self-signed certificate (which is the only thing that I can produce easily) will not change the behavior of Eclipse during installation: The confirmation dialog will still appear, only a real certificate with a certificate chain to a root authority would actually avoid that dialog. Therefore I'm closing this, since the effort will not really have benefit for the users (except for showing the self-signed certificate).

Some links for interested readers:
* https://nirmalsasidharan.wordpress.com/2010/09/04/signing_eclipse_plugins/
* https://dev.to/kengotoda/sign-eclipse-plugin-by-gradle-4md8
